src/components/DepositWidget/index.tsx

```diff
@@ -72,6 +99,152 @@ const DepositWidget: React.FC = () => {
+    } catch (error) {
+      console.error('Error depositing', error)
+      toast.error(`Error depositing: ${error.message}`)
+    }
+  }
+
+  async function _requestWithdraw(amount: BN, tokenAddress: string): Promise<void> {
+    const { symbol, decimals } = _getToken(tokenAddress)
+    try {
+      log(`Processing withdraw request of ${amount} ${symbol} from ${userAddress}`)
+
+      const result = await depositApi.requestWithdraw(userAddress, tokenAddress, amount, txOptionalParams)
+      log(`The transaction has been mined: ${result.receipt.transactionHash}`)
+
+      if (mounted.current) {
+        _updateToken(tokenAddress, otherParams => {
+          return {
+            ...otherParams,
+            withdrawingBalance: amount,
+            claimable: false,
+            highlighted: true,
+          }
+        })
+        _clearHighlight(tokenAddress)
+      }
+
+      toast.success(`Successfully requested withdraw of ${formatAmount(amount, decimals)} ${symbol}`)
+    } catch (error) {
+      console.error('Error requesting withdraw', error)
+      toast.error(`Error requesting withdraw: ${error.message}`)
+    }
+  }
+
+  async function _claim(tokenAddress: string): Promise<void> {
+    const { withdrawingBalance, symbol, decimals } = _getToken(tokenAddress)
+    try {
+      console.debug(`Starting the withdraw for ${formatAmountFull(withdrawingBalance, decimals)} of ${symbol}`)
+      _updateToken(tokenAddress, otherParams => {
+        return {
+          ...otherParams,
+          claiming: true,
+        }
+      })
+      const result = await depositApi.withdraw(userAddress, tokenAddress, txOptionalParams)
+
+      if (mounted.current) {
+        _updateToken(tokenAddress, ({ exchangeBalance, walletBalance, ...otherParams }) => {
+          return {
+            ...otherParams,
+            claiming: false,
+            exchangeBalance: exchangeBalance.sub(withdrawingBalance),
+            withdrawingBalance: ZERO,
+            claimable: false,
+            walletBalance: walletBalance.add(withdrawingBalance),
+            highlighted: true,
+          }
+        })
+        _clearHighlight(tokenAddress)
+      }
+
+      log(`The transaction has been mined: ${result.receipt.transactionHash}`)
+      toast.success(`Withdraw of ${withdrawingBalance} ${symbol} completed`)
+    } catch (error) {
+      console.error('Error executing the withdraw request', error)
+      toast.error(`Error executing the withdraw request: ${error.message}`)
+    }
+  }
+
+  async function _enableToken(tokenAddress: string): Promise<void> {
+    const { symbol } = _getToken(tokenAddress)
+    try {
+      _updateToken(tokenAddress, otherParams => {
+        return {
+          ...otherParams,
+          enabling: true,
+        }
+      })
+      const result = await erc20Api.approve(
+        tokenAddress,
+        userAddress,
+        contractAddress,
+        ALLOWANCE_MAX_VALUE,
+        txOptionalParams,
+      )
```

Comment:
Note to self: I changed the order of this function to be consistent with others and have the `from`/`senderAddress` as first parameter https://github.com/gnosis/dex-react/pull/110/files#diff-db4249d04c7a8259d80ed280f6bfef22L6

No need to change it in this PR, unless that PR is merged first.

Reply:
I think it's not merged for now. So resolving this one